src/libs/vtkh/filters/UniformGrid.cpp

```diff
@@ -0,0 +1,759 @@
+      }
+
+      return res;
+    }
+  }; //struct reduceFields
+
+};//class globalReduceField
+#endif
+
+} //namespace detail
+
+UniformGrid::UniformGrid()
+	: m_invalid_value(std::numeric_limits<double>::min())
+{
+
+}
+
+UniformGrid::~UniformGrid()
+{
+
+}
+
+void
+UniformGrid::PreExecute()
+{
+  Filter::PreExecute();
+}
+
+void
+UniformGrid::DoExecute()
+{
+#ifdef VTKH_PARALLEL
+  // Setup VTK-h and VTK-m comm.
+  MPI_Comm mpi_comm = MPI_Comm_f2c(vtkh::GetMPICommHandle());
+  vtkm::cont::EnvironmentTracker::SetCommunicator(vtkmdiy::mpi::communicator(vtkmdiy::mpi::make_DIY_MPI_Comm(mpi_comm)));
+  int par_rank;
+  int par_size;
+  MPI_Comm_rank(mpi_comm, &par_rank);
+  MPI_Comm_size(mpi_comm, &par_size);  
+#endif
+  this->m_output = new DataSet();
+
+  std::vector<vtkm::Id> domain_ids = this->m_input->GetDomainIds(); 
+  const int num_domains = domain_ids.size();
+  //add mask to keep track of valid points after sampling
+  this->m_input->AddConstantPointField(0.0, "mask");
+
+  //put vtkm datasets into a partitionedDS for vtkm::Merge
+  vtkm::cont::PartitionedDataSet sampled_doms;
+  for(int i = 0; i < num_domains; ++i)
+  {
+    vtkm::cont::DataSet dom;
+    
+    if(this->m_input->HasDomainId(domain_ids[i]))
+    {
+      dom = this->m_input->GetDomainById(domain_ids[i]);
+
+      sampled_doms.AppendPartition(dom);
+    }
+  }
+#ifdef VTKH_PARALLEL
+  //if there is no data, add some empty 
+  vtkm::cont::DataSet tmp_empty;
+  if(sampled_doms.GetNumberOfPartitions() == 0)
+  {
+    tmp_empty = vtkm::cont::DataSetBuilderUniform::Create(m_dims, m_origin, m_spacing);
+    bool valid_field;
+    vtkm::cont::Field::Association assoc = this->m_input->GetFieldAssociation(m_field, valid_field);
+
+    if(!valid_field)
+    {
+      this->m_output = this->m_input;
+      return;
+    }
+    
+    vtkm::Id field_id = this->m_input->GetFieldType(m_field, valid_field);
+    vtkm::cont::Field empty_field = vtkh::detail::MakeEmptyField(m_field,field_id,m_dims,assoc);
+    tmp_empty.AddField(empty_field);
+    sampled_doms.AppendPartition(tmp_empty);
+  }
+#endif
+
+  ///
+  /// Approach we can use that would not need MergeDataSets:
+  ///
+  /// (This will also be more memory efficient b/c we don't need to convert
+  /// everything to a fused unstructured grid)
+  ///
+  /// create local output grid `local_res` (including masking info)
+  /// create global output grid `global_res` (including masking info)
+  /// create tmp local output grid `local_res_tmp` (including masking info)
+  ///
+  /// for each domain `d`:
+  ///   reset local_res_tmp (including making info)
+  ///   execute probe filter on `d` with output in `local_res_tmp`
+  ///   combine results from `local_res_tmp` into `local_res`
+  /// if mpi parallel:
+  ///.  Use global reduce w/ `local_res` to create `global_res`
+  ///
+  ///.  (We know that all ranks will have something to reduce, b/c
+  ///.   even if they have no domains, the still created the local
+  ///    output grid)
```

Comment:
@nicolemarsaglia  see note here of how we can accomplish this w/o MergeDataset -- allowing us to use vtk-m 2.1
Also, this would be more memory efficient b/c merging will create a fused unstructured grid which can require a lot more space if the input was uniform (explicit coords and connectivity, maybe copying and repacking of field arrays)